File: 2049-eliminate-maximum-number-of-monsters/2049-eliminate-maximum-number-of-monsters.py
class Solution:
    def eliminateMaximum(self, dist: List[int], speed: List[int]) -> int:
        timeToCity = [dist[i]/speed[i] for i in range(len(dist))]

        timeToCity.sort()

        for i in range(len(dist)):
            if timeToCity[i] <= i:
                return i

        return len(dist)

File: 2049-eliminate-maximum-number-of-monsters/2049-eliminate-maximum-number-of-monsters.java
class Solution {
    public int eliminateMaximum(int[] dist, int[] speed) {
        int n = dist.length;
        double[] timeToCity = new double[n];

        for(int i = 0; i < n; i++){
            timeToCity[i] = (double)dist[i]/speed[i];
        }

        Arrays.sort(timeToCity);

        for(int i = 0; i < n; i++){
            if(timeToCity[i] <= i) return i;
        }

        return n;
    }
}